File: mobile/ios/packages/town-crier-domain/Sources/ValueObjects/LatestUnreadEvent.swift
import Foundation

/// Per-application unread-notification descriptor surfaced on each row of the
/// applications-by-zone result. Drives the saturated/muted styling of the row's
/// `ApplicationStatusPill` and the `recent-activity` sort on the Applications
/// screen. `nil` (or absent) when the application has no unread notification
/// for the user (server-side `read_at IS NULL`).
///
/// `type` mirrors the API's `NotificationEventType` discriminator, currently
/// `"NewApplication"` or `"DecisionUpdate"`. Carried as a string rather than
/// an enum so unrecognised future values do not cause decoding failures.
/// `decision` is the raw PlanIt `app_state` (e.g. `"Permitted"`) for
/// `DecisionUpdate` events; `nil` otherwise. `createdAt` is the instant the
/// notification was raised — used by the `recent-activity` sort.
///
/// See ADR 0035 (`docs/adr/0035-per-application-notification-read-state.md`).
public struct LatestUnreadEvent: Equatable, Sendable {
  public let type: String
  public let decision: String?
  public let createdAt: Date

  public init(type: String, decision: String?, createdAt: Date) {
    self.type = type
    self.decision = decision
    self.createdAt = createdAt
  }
}

File: api/src/town-crier.application/PlanningApplications/LatestUnreadEvent.cs
using TownCrier.Domain.Notifications;

namespace TownCrier.Application.PlanningApplications;

/// <summary>
/// Per-application unread-notification descriptor surfaced on each row of the
/// applications-by-zone result. Drives the saturated colour and copy of the
/// status pill in iOS/web. <c>null</c> when no notification exists strictly
/// after the user's <c>lastReadAt</c> watermark, or the user has no watermark
/// document yet (first-touch path; clients will seed via
/// <c>GET /v1/me/notification-state</c>).
/// </summary>
/// <remarks>
/// See <c>docs/specs/notifications-unread-watermark.md#api-augment-applications</c>.
/// </remarks>
/// <param name="Type">The lifecycle event the notification was raised for.</param>
/// <param name="Decision">The raw PlanIt decision string (e.g. "Permitted") for <see cref="NotificationEventType.DecisionUpdate"/>; null otherwise.</param>
/// <param name="CreatedAt">The instant the notification was raised. Used by the iOS push-tap path to advance the watermark.</param>
public sealed record LatestUnreadEvent(
    NotificationEventType Type,
    string? Decision,
    DateTimeOffset CreatedAt);
